### DISCUSSION TITLE
Creating coupled history files

### DISCUSSION THREAD

**Discussion by adagj** [NorESMhub/noresm3_dev_simulations#50] [Spin up]
Hi,
we had a meeting today where we discussed spin up of BLOM. We want to generate coupled history files in the upcomming simulation (only 2 years) to test and document the spin up procedure. Do we want history files for land and ice sheet in that simulation? The model is totally out of balance, so this is for testing and documenting. Let me know what you think @rosiealice @mpetrini-norce @hgoelzer

**Comment by rosiealice** [comment]
I'm not totally sure I have a coherent answer to the question without knowing more about the process you discussed. I guess it wouln't hurt to have those baseline files, as you note, for the purposes of documentation of where we are now...

**Reply by hgoelzer** [reply]
Just echoing Rosie's reply from the ice sheet side. Testing sounds good, but not sure how exactly this would look like and what it would require from us.

**Reply by tjiputra** [reply]
We need to know what (atm and land) fields we need to save to spin up the icesheet offline.

**Reply by micpetrini** [reply]
> We need to know what (atm and land) fields we need to save to spin up the icesheet offline.

One thing that's great for us is to set "histaux_l2x1yrg = .true." in the cpl namelist (see also https://github.com/NorESMhub/CTSM/issues/82). This enables us to calculate the ice sheet forcing offline in the exact same way NorESM would do in a coupled run with ice sheets active. There is also CLM output that can be used to calculate ice sheet forcing using different methods, @hgoelzer can you share your list of variables and output frequency?

**Comment by mvertens** [comment]
@hgoelze -  @tjiputra will need to repeat the two year run in #52  with new mediator history fields since the original auxiliary file variables are no longer all correct out of the box.  I've have updated the list to include `histaux_l2x1yrg = .true.`.  
- The aim of #52 is to show what fields are needed to be output by the mediator and then used by DATM (for clm spinup), DATM/DROF (for blom spinup) and DLND (for cism spinup).
- Are there new settings needed for user_nl_clm that are needed to calculate ice sheet forcing using different methods? If these are needed for a cplhist simulation - then I'm assuming these would need to be read by dlnd? Can we clarify this further. 
- At this point I would suggest just using `histaux_l2x1yrg = .true. `to demonstrate the mediator history files needed for spinups.

**Reply by hgoelzer** [reply]
Here is what we had in user_nl_clm

hist_nhtfrq = 0, -24, 0, -8760
hist_mfilt  = 1, 5, 1, 1

hist_fincl1 = 'FERT_TO_SMINN','NFIX_TO_SMINN','LITFIRE','LITR1C_TO_SOIL1C','LITR2C_TO_SOIL1C','LITR3C_TO_SOIL2C','M_LEAFC_TO_LITTER','M_FROOTC_TO_LITTER','M_LIVESTEMC_TO_LITTER','M_DEADSTEMC_TO_LITTER','M_LIVECROOTC_TO_LITTER','M_DEADCROOTC_TO_LITTER','FIRA', 'FIRE_ICE', 'FSH_ICE', 'EFLX_LH_TOT_ICE', 'QSNOMELT_ICE', 'QSNOFRZ_ICE', 'QSOIL_ICE', 'QICE', 'QICE_MELT', 'FSA', 'FSR_ICE', 'TOPO_COL_ICE', 'FSDS', 'FLDS', 'LWdown', 'RAIN_ICE', 'SNOW_ICE', 'TSA_ICE', 'TG_ICE', 'H2OSNO_ICE', 'ICE_MODEL_FRACTION'

hist_fincl2 = 'QRUNOFF', 'SOILLIQ', 'SOILICE', 'SOILWATER_10CM', 'TSA', 'TSL', 'GPP', 'AR', 'HR'


hist_fincl3 = 'FIRA', 'FIRE', 'FSH', 'EFLX_LH_TOT', 'QSNOMELT', 'QSNOFRZ', 'QSOIL', 'QICE', 'QICE_MELT', 'FSA', 'FSR', 'TOPO_COL', 'FSDS', 'FLDS', 'LWdown', 'RAIN', 'SNOW', 'TSA', 'TG', 'H2OSNO'

hist_fincl4 = 'PCT_LANDUNIT', 'PCT_NAT_PFT', 'FSNO_ICE', 'PCT_CFT', 'PCT_LANDUNIT', 'PCT_NAT_PFT', 'PCT_GLC_MEC', 'EFLX_DYNBAL', 'QFLX_LIQ_DYNBAL', 'QFLX_ICE_DYNBAL'

hist_dov2xy = .true., .true., .false., .true.

This additional information was part of the standard keyclim experiments. I don't know which if this still work. And if we need them all, but it could be a good starting point. 
Sorry I don't have better insight at this point. Heiko
